text/0088-tvm-update.md

```diff
@@ -0,0 +1,142 @@
+- **TEP**: [88](https://github.com/ton-blockchain/TEPs/pull/87)
+- **title**: TVM update
+- **status**: Draft
+- **type**: Core
+- **authors**: [@SpyCheese](https://github.com/SpyCheese) [@EmelyanenkoK](https://github.com/EmelyanenkoK) 
+- **created**: 07.07.2022
+- **replaces**: -
+- **replaced by**: -
+
+# Summary
+
+This proposal suggests to
+- add new op-codes to TVM related to block data retrieving, hashing, cryptographic signing and message sending
+- include into `c7` register of TVM smart-contract' own code, additional data on transaction and information about masterchain blocks
+ 
+# Motivation
+
+Current TVM implementation makes it hard to implement some common thing. Proposal aims to make it easier and cheaper to:
+- hash big chanks of data (including non-sha256 hashes)
+- work with Bitcoin/Ethereum compatible signtatures
+- prove TON onchain data
+- calculate fee for sending messages
+
+# Guide
+
+Draft implementation is available [here](https://github.com/SpyCheese/ton/commits/new-opcodes).
+
+This proposal suggests to 
+1. Extend **c7** tuple from 10 to 14 elements:
+   - **10**: code of the smart contract.
+   - **11**: value of the incoming message.
```

Comment:
Currently we have it on stack when processing internal message, and it is ok, i don't think we need to move it to `c7` at least because we have it only for internal messages, stack looks pretty fine. Less universal and not so useful imo.